algebra/group_power.lean

```diff
@@ -471,6 +471,22 @@ theorem pow_nonneg {a : α} (H : 0 ≤ a) : ∀ (n : ℕ), 0 ≤ a ^ n
 | 0     := zero_le_one
 | (n+1) := mul_nonneg H (pow_nonneg _)
 
+theorem pow_lt {x y : α} {n : ℕ} (Hxy : x < y) (Hxpos : 0 < x) (Hnpos : 0 < n) : x ^ n < y ^ n :=
+begin
+  rw ←nat.sub_add_cancel Hnpos,
+  induction (n - 1), { simpa only [pow_one] },
+  rw [pow_add, pow_add, nat.succ_eq_add_one, pow_one, pow_one],
+  apply mul_lt_mul ih (le_of_lt Hxy) Hxpos (le_of_lt (pow_pos (lt_trans Hxpos Hxy) _))
+end
+
+theorem pow_eq {x y : α} {n : ℕ} (Hxpos : 0 < x) (Hypos : 0 < y) (Hnpos : 0 < n) (Hxyn : x ^ n = y ^ n) : x = y :=
```

Comment:
`pow_right_inj` is a better name here.

Reply:
This should hold under the weaker assumption that x and y are nonnegative, right?